### DISCUSSION TITLE
French translation

### DISCUSSION THREAD

**Discussion by marcpley** [pawelsalawa/sqlitestudio#5075] [General]
Hello. I'm new user of SQLite and SQLiteStudio.
I have choose SQLiteStudio after testing  DBBrowser, dBeaver, BeeKeeper, Navicat and LibreOffice Base.
Great thanks for your App Pawel !

I could help on the french translation. I'm native french speaker. How to contribute ? I'm new to github too ;-)

Here there is a bad translation :
https://github.com/pawelsalawa/sqlitestudio/blob/819b54e96023619846fcd55d6b98efe9ec26396f/SQLiteStudio3/guiSQLiteStudio/translations/guiSQLiteStudio_fr_FR.ts#L7025

Need to replace the word "charher" with "charger". I'm totally sure of that.

Not sure but : generally, the french translation of "Table" (if used for a database) is not "Tableau" but "Table" (same as in english). This is important because terribly visible when you open the app for the first time !
But I see that you use "Column" instead of "Field" for the English translation, that seem to me strange.
In french we use "Champ" for data field and "Colonne" for a column in a grid, it's not the same thinks. A column displays the data of a table field.

**Comment by pawelsalawa** [comment]
Instructions for translators are at https://github.com/pawelsalawa/sqlitestudio/wiki/Language_Translations Thanks!

Google Translator translates "data table" to "tableau de données". Does the "Table"/"Tableau" depend on the context in which you translate it? Can you explain difference between "Table" and "Tableau"?

**Comment by marcpley** [comment]
"Tableau" is grid, a manner to present data. In a grid (Tableau) there is columns (colonnes) and rows (lignes).
"Table" is something to store data in (this was true before the computers exists). The structure of the table consist in fields (champs). The data of table consist in records (enregistrements).
I agree that "row" is currently used instead of "record" in English. 
In french "colonne" is sometime used instead of "champ" (fields), for both the data structure and the data display, but it's considered as not good in the french wikipedia article. Idem for "ligne" (row) used instead of "enregistrement" (record).
No one  uses  "Tableau" (grid) instead of "Table".

So, a minimum would be replacing "Tableau" by "Table".

A more complete action would be replacing:
"Colonne" by "Field" if about a table structure.

And eventually:
 "Table" (Grid view) by "Tableau"

In the config windows there is also stranges or wrongs translations. I can check that if you want.

**Comment by pawelsalawa** [comment]
I see. Feel invited to contribute. Thank you!

**Comment by marcpley** [comment]
Done, only replacing "Tableau" with "Table" in structure context.
Other modifications (champ vs colonne) may cause more damage than increase.
I also correct some complex English sentences that was completly incomprehensible in French.
If a beta of 3.4.5 can be see, I would verify my work.

**Comment by marcpley** [comment]
There is no translation entry for 'Execute "EXPLAIN" query' (in Settings, shortcuts)

**Reply by pawelsalawa** [reply]
The string to translate is `Execute "%1" query`, where %1 is replaced with SQL command, because SQL keywords are not entities to be translated, so you can treat them as "something", like the "%1" above.

**Reply by marcpley** [reply]
Sorry, I don't find this entry.

**Reply by pawelsalawa** [reply]
Right, that's because the `"` is not written as `"` in translation files, so the string you're looking for is `Execute &quot;%1&quot; query`.

**Reply by marcpley** [reply]
Ok, thanks :-)

**Comment by marcpley** [comment]
I hardly understand this entry : Move keyboard input focus to the results view below
Do you mean "Move keyboard input focus below the results view"
or "Move keyboard input focus to the below results view"

**Reply by pawelsalawa** [reply]
It's "Move keyboard input focus to the below results view".

The thing is that back in old times, the results view could only be "below". Nowadays it can also be in a separate tab, so not "below". It is not critical and I don't know if there is any better solution for this. I don't think it's a good idea to add more text to explain "applicable only if results are displayed below and not in separate tab".

**Reply by marcpley** [reply]
Thanks, I translate that.

**Comment by marcpley** [comment]
In config dialog, shortcuts, the names of 5 firsts (in english) sections are like "A xxx window".
This don't seem to me coherent with the others section names. Example: "Data form view" (not "A data form view").
And if not removing this "A", the strings would be like "In a xxx window".

Other point: sections are alpha sorted, so  the order is not the same depending of UI language. A logical order, coherent with menus orders or coherent with database logic would be better.

Those are non important issues. Pawel, I don't know if you are interested in such remarks or if they waste your time. Feel free to tell me "work on more important things" or somethng like that ;-)

**Reply by pawelsalawa** [reply]
It's a problem of amount of time I can dedicate to the project (which is extremely little), so I will be frank and say I won't address this anytime soon, as indeed there are more important matters/bugs/enhancements.

**Comment by marcpley** [comment]
Last french untranslated string are about collation. But already translated strings use 3 french terms : Regroupement, Classement, Collation.
I would uniformize but I don't understand enougth what is used for the Collation editor. Could you give an example of a collation that can be set in the collation editor ?

**Reply by pawelsalawa** [reply]
Collations in SQL are basically an algorithm using which strings are compared with each other and sorted. SQLite has doc entry on it: https://sqlite.org/datatype3.html#collation
Collations are used for example when you use `ORDER BY` clause.

SQLite support creating your own, custom collation algorithms for comparing values with each other.
SQLiteStudio provides handy way to create these collations - in Collations Editor window.

**Comment by marcpley** [comment]
Hello Pawel
Will the update of french translations be in the 3.5.0 release ? I don't see them in 3.4.5.

**Reply by pawelsalawa** [reply]
Yes, crowdin translations currently work only for master branch, i.e. now for 3.5.0.
